publisher-sdk/src/main/java/com/criteo/publisher/privacy/UserPrivacyUtil.java

```diff
@@ -148,4 +152,42 @@ public void storeMopubConsent(@Nullable String mopubConsent) {
   public String getMopubConsent() {
     return safeSharedPreferences.getString(MOPUB_CONSENT_SHARED_PREFS_KEY, "");
   }
+
+  /**
+   * Indicate if user didn't give its consent for CSM when applicable.
+   *
+   * <ul>
+   *   <li>
+   *     IABTCF_PublisherRestrictions{ID}: publisher restrictions on purpose {ID} (only for ID=1)
+   *     <ul>
+   *       <li>if RestrictionType = 0: Purpose Flatly Not Allowed by Publisher => <b>No CSM</b></li>
+   *       <li>if RestrictionType = 1: Consent is required > Ignore restriction > fallback on other consent checks</li>
+   *       <li>if RestrictionType = 2: Legitimate interest is required > <b>No CSM</b></li>
+   *     </ul>
+   *   </li>
+   *   <li>
+   *     IABTCF_VendorConsents AND IABTCF_VendorLegitimateInterests:
+   *     <ul>
+   *       <li>VendorConsent: Check Bitfield OR Range whether Criteo (91) == 1</li>
+   *       <li>VendorLegitimateInterest: Check Bitfield OR Range whether Criteo (91) == 1</li>
+   *       <li>If Criteo (91) is not set to 1 in either VendorConsent or VendorLegitimateInterest: <b>No CSM</b></li>
+   *     </ul>
+   *   </li>
+   * </ul>
+   */
+  public boolean isCsmDisallowed() {
+    TcfGdprStrategy tcfGdprStrategy = tcfStrategyResolver.resolveTcfStrategy();
+    if (tcfGdprStrategy == null) {
+      return false;
+    }
+
+    PublisherRestrictionType restrictionTypeForPurpose1 = tcfGdprStrategy.getPublisherRestrictionTypeForPurpose1();
+    if (restrictionTypeForPurpose1 == NOT_ALLOWED || restrictionTypeForPurpose1 == REQUIRE_LEGITIMATE_INTEREST) {
```

Comment:
If legitimate interest is required, shouldn't we check if legitimate interest  is established for `Criteo`? (`IABTCF_VendorLegitimateInterests`)

Reply:
From the TCF point of view we're vendor, not publishers. Hence there is no link between the two.
For the publisher restriction, it would be like: "do you consent that a publisher store a personal data". Vs for the vendor it would be: "do you consent that a publisher is sending data to a vendor for ...".

I guess it's complicated legal requirements, and to be honest, I prefer to trust product team on this 😄